openedx/features/idea/models.py

```diff
@@ -2,5 +2,77 @@
 from __future__ import unicode_literals
 
 from django.db import models
+from django.contrib.auth.models import User
+from django.core.validators import FileExtensionValidator
+from django_countries.fields import CountryField
 
-# Create your models here.
+from functools import partial
+
+from lms.djangoapps.onboarding.models import Organization
+from .helpers import upload_to_path
+
+
+class Location(models.Model):
+    country = CountryField()
+    city = models.CharField(max_length=255)
+
+    @property
+    def location(self):
+        return '{city}, {country}'.format(city=self.city, country=self.country)
+
+    def __unicode__(self):
+        return self.location
+
+    class Meta:
+        abstract = True
+
+
+class VisualAttachment(models.Model):
+
+    video_link = models.URLField(blank=True, null=True)
+    image = models.FileField(
+        upload_to=partial(upload_to_path, folder='images'), blank=True, null=True,
+        validators=[FileExtensionValidator(['jpg', 'png', 'pdf'])],
+        help_text='Accepted extensions: .jpg, .png, .pdf'
+    )
+    file = models.FileField(
+        upload_to=partial(upload_to_path, folder='files'), blank=True, null=True,
+        validators=[FileExtensionValidator(['docx', 'pdf', 'txt'])],
+        help_text='Accepted extensions: .docx, .pdf, .txt'
+    )
+
+    def __unicode__(self):
+        return self.video_link
+
+    class Meta:
+        abstract = True
+
+
+class OrganizationBase(models.Model):
+    organization = models.ForeignKey(
+        Organization,
+        related_name='%(app_label)s_%(class)ss',
```

Comment:
I am not sure about best practice, but do you think to hardcode it would be more readable?

Reply:
This is the best and recommend way as per [django](https://docs.djangoproject.com/en/1.11/topics/db/models/#be-careful-with-related-name-and-related-query-name)